lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -770,13 +773,63 @@ struct ConvertHWToBTOR2Pass
     // Genrate the bad btor2 intruction
     genBad(op);
   }
+
   // Assumptions are converted to a btor2 constraint instruction
   void visitSV(sv::AssumeOp op) {
     // Extract the expression that we want our constraint to be about
     Value expr = op.getExpression();
     genConstraint(expr);
   }
 
+  // Assertions are sometimes encoded using a combination of fatal and error, we
+  // thus need to find the parenting condition and generate a bad statement
+  // using that.
+  void visitEncodedAssert(Operation *op) {
+    genSort("bitvec", 1);
+
+    // Fatal simply fails, so we need to find in which case we are failing
+    // For some reason Chisel assertions are sometimes encoded using a fatal and
+    // an error.
+    if (auto ifop = dyn_cast<sv::IfOp>(((Operation *)op)->getParentOp())) {
+      Value cond = ifop.getOperand();
+
+      // Check that the operand is a condition and not an sv macro, otherwise
+      // look for another parent. Some functions like printf will be converted
+      // using additional macros as first level conditions to allow for more
+      // error-reporting control later down the road. To do this we simply make
+      // sure that it's not part of the sv dialect, given that at this point a
+      // condition should not be expressed using operations from that dialect.
+      if (cond.getDefiningOp()->getDialect() == op->getDialect()) {
```

Comment:
It crashes when cond is a BlockArgument.  Also even when the dialect is same(SV), we still need to check the op is MacroRef.

Reply:
Is this really the only situation we can run into ? Can't it also be defined using other sv constructs when lowering from Chisel?